core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -172,6 +173,34 @@ public static ByteString fromBase64(String base64) {
     }
   }
 
+  /** SQL MD5(string) function. */
+  public static @Nonnull String md5(@Nonnull String string)  {
+    return md5_(string.getBytes(UTF_8));
+  }
+
+  /** SQL MD5(string) function for binary string. */
+  public static @Nonnull String md5(@Nonnull ByteString string)  {
+    return md5_(string.getBytes());
+  }
+
+  private static @Nonnull String md5_(byte[] bytes)  {
+    return DigestUtils.md5Hex(bytes);
+  }
```

Comment:
We can go without private `md5_` and  `sha1_` method, since they are just simple function call. You can call them directly in the function.

Reply:
Hi @hsuanyi , thanks for your replies. I have upgraded my codes as your advise.